src/core/bau_inputs.jl

```diff
@@ -164,15 +175,134 @@ function BAUInputs(p::REoptInputs)
+    # that accounts for the PV capacity degradation over the analysis_years. In other words, by
+    # multiplying the pv production_factor by the levelization_factor we are modeling the average pv production.
+    bau_grid_to_load = copy(s_bau.electric_load.loads_kw)
+    bau_grid_to_load_critical = copy(s_bau.electric_load.critical_loads_kw)
+    for pv in p.s.pvs if pv.existing_kw > 0
+        bau_grid_to_load .-= p.levelization_factor[pv.name] * pv.existing_kw * p.production_factor[pv.name, :].data
+    end end
+    for pv in p.s.pvs if pv.existing_kw > 0
+        bau_grid_to_load_critical .-= p.levelization_factor[pv.name] * pv.existing_kw * p.production_factor[pv.name, :].data
+    end end
+
+    #No grid emissions, or pv exporting to grid, during an outage
+    if p.s.electric_utility.outage_start_time_step != 0 && p.s.electric_utility.outage_end_time_step != 0
+        for i in range(p.s.electric_utility.outage_start_time_step, stop=p.s.electric_utility.outage_end_time_step)
+            bau_grid_to_load[i] = 0
+        end
+    end
+
+    #If no net emissions accounting, no credit for RE grid exports:
+    if !p.s.site.include_exported_elec_emissions_in_total
+        bau_grid_to_load = [max(i,0) for i in bau_grid_to_load]
+    end
+
+    bau_grid_emissions_lb_CO2_per_year = sum(p.s.electric_utility.emissions_factor_series_lb_CO2_per_kwh .* bau_grid_to_load) / p.s.settings.time_steps_per_hour
+    bau_emissions_lb_CO2_per_year += bau_grid_emissions_lb_CO2_per_year
+
+    bau_grid_emissions_lb_NOx_per_year = sum(p.s.electric_utility.emissions_factor_series_lb_NOx_per_kwh .* bau_grid_to_load) / p.s.settings.time_steps_per_hour
+    bau_emissions_lb_NOx_per_year += bau_grid_emissions_lb_NOx_per_year
+
+    bau_grid_emissions_lb_SO2_per_year = sum(p.s.electric_utility.emissions_factor_series_lb_SO2_per_kwh .* bau_grid_to_load) / p.s.settings.time_steps_per_hour
+    bau_emissions_lb_SO2_per_year += bau_grid_emissions_lb_SO2_per_year
+
+    bau_grid_emissions_lb_PM25_per_year = sum(p.s.electric_utility.emissions_factor_series_lb_PM25_per_kwh .* bau_grid_to_load) / p.s.settings.time_steps_per_hour
+    bau_emissions_lb_PM25_per_year += bau_grid_emissions_lb_PM25_per_year
+
+    ## Generator emissions (during outages)
+    if "Generator" in p.techs.all
+        bau_emissions_lb_CO2_per_year += generator_fuel_use_gal * p.s.generator.emissions_factor_lb_CO2_per_gal
+        bau_emissions_lb_NOx_per_year += generator_fuel_use_gal * p.s.generator.emissions_factor_lb_NOx_per_gal
+        bau_emissions_lb_SO2_per_year += generator_fuel_use_gal * p.s.generator.emissions_factor_lb_SO2_per_gal
+        bau_emissions_lb_PM25_per_year += generator_fuel_use_gal * p.s.generator.emissions_factor_lb_PM25_per_gal
+    end
+
+    ## Boiler emissions
+    if "ExistingBoiler" in p.techs.all
+        for heat_type in ["space_heating", "dhw"] #TODO: are these heating types the same? can we get them from s?
```

Comment:
do we need to address this TODO?

Reply:
I don't remember adding this TODO. These are the correct heat load types though, and no we don't save a list of them in s, so I'll delete. We could maybe do something similar to the tech category lists we create, but I don't see enough value to justify spending time on it right now.